sdk/cosmos/azure_data_cosmos/src/models/container_properties.rs

```diff
@@ -0,0 +1,402 @@
+use std::time::Duration;
+
+use azure_core::Model;
+use serde::{Deserialize, Deserializer};
+
+use crate::models::SystemProperties;
+
+#[cfg(doc)]
+use crate::clients::ContainerClientMethods;
+
+fn deserialize_ttl<'de, D>(deserializer: D) -> Result<Option<Duration>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    Ok(Option::<u64>::deserialize(deserializer)?.map(Duration::from_secs))
+}
+
+/// Properties of a Cosmos DB container.
+///
+/// Returned by [`ContainerClient::read()`](crate::clients::ContainerClient::read()).
+#[non_exhaustive]
+#[derive(Model, Clone, Debug, Deserialize, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+pub struct ContainerProperties {
+    /// The ID of the container.
+    pub id: String,
+
+    /// The time-to-live for items in the container.
+    ///
+    /// For more information see <https://docs.microsoft.com/azure/cosmos-db/time-to-live#time-to-live-configurations>
+    #[serde(default)]
+    #[serde(deserialize_with = "deserialize_ttl")]
+    pub default_ttl: Option<Duration>,
+
+    /// The time-to-live for the analytical store in the container.
+    ///
+    /// For more information see <https://docs.microsoft.com/azure/cosmos-db/analytical-store-introduction#analytical-ttl>
+    #[serde(default)]
+    #[serde(deserialize_with = "deserialize_ttl")]
+    pub analytical_storage_ttl: Option<Duration>,
+
+    /// The definition of the partition key for the container.
+    pub partition_key: PartitionKeyDefinition,
+
+    /// The indexing policy for the container.
+    pub indexing_policy: Option<IndexingPolicy>,
+
+    /// The unique key policy for the container.
+    pub unique_key_policy: Option<UniqueKeyPolicy>,
+
+    /// The conflict resolution policy for the container.
+    pub conflict_resolution_policy: Option<ConflictResolutionPolicy>,
+
+    /// A [`SystemProperties`] object containing common system properties for the container.
+    #[serde(flatten)]
+    pub system_properties: SystemProperties,
+}
+
+/// Represents the partition key definition for a container.
+#[non_exhaustive]
+#[derive(Clone, Debug, Deserialize, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+pub struct PartitionKeyDefinition {
+    /// The list of partition keys paths.
+    pub paths: Vec<String>,
+
+    /// The version of the partition key hash in use.
+    pub version: Option<usize>,
```

Comment:
Is version truly unbounded by Cosmos itself? Though, perhaps this doesn't matter since it's only for deserialization but would also seem to put more burden on clients since [numbers default to i32](https://doc.rust-lang.org/book/ch03-02-data-types.html#:~:text=integer%20types%20default%20to%20i32) if no other signal is provided.

Reply:
It's not really unbounded, [the .NET SDK does use an enum for this](https://learn.microsoft.com/en-us/dotnet/api/microsoft.azure.cosmos.partitionkeydefinitionversion?view=azure-dotnet). I was using a number primarily to avoid breaking deserialization later when a new version is returned (though we have other enums that will probably break when new values are used) and because it is intended to be a simple integer version number. The Go SDK uses a bare integer, but Go doesn't really have enums anyway.

I'll make this an enum for now. It does raise the idea of having a way to represent "Unknown" values in all of our enums, but that's a larger issue across the rest of Cosmos (and likely other SDKs)

Reply:
After digging a bit in to serde's support for enums-as-numbers, I think I'm going to keep this a number for now. Serde doesn't directly support enums-as-numbers, it requires using the [`serde_repr`](https://crates.io/crates/serde_repr) crate, or a custom `Deserialize` implementation. Adding the dependency isn't a dealbreaker, but given the desire to potentially support new versions without updating the client, I think it's reasonable to leave this a number for now. I did change it to `i32` to match with Rust's default for numbers.

Reply:
I'd rather not take more dependencies as well. My main concern was about the integer type itself, but if you and your team think that most comparisons would be done in a way where rustc can infer a number as a `usize` because it's only used in a comparison or something for this use case, that's fine.

Reply:
I don't think we need to future proof this too much.  We're only on hash v2 and we're 10 years into this thing ;)

Reply:
So you're saying we're due ;P